src/modi/c13.rs

```diff
@@ -864,53 +909,85 @@ impl<'a> FallibleIterator for C13FileIterator<'a> {
 }
 
 pub struct C13LineProgram<'a> {
-    data: &'a [u8],
     file_checksums: DebugFileChecksumsSubsection<'a>,
+    line_sections: Vec<DebugLinesSubsection<'a>>,
 }
 
 impl<'a> C13LineProgram<'a> {
     pub(crate) fn parse(data: &'a [u8]) -> Result<Self> {
-        let checksums_data = DebugSubsectionIterator::new(data)
-            .find(|sec| sec.kind == DebugSubsectionKind::FileChecksums)?
-            .map(|sec| sec.data);
+        let mut file_checksums = DebugFileChecksumsSubsection::default();
+        let mut line_sections = Vec::new();
 
-        let file_checksums = match checksums_data {
-            Some(d) => DebugFileChecksumsSubsection::parse(d)?,
-            None => DebugFileChecksumsSubsection::default(),
-        };
+        let mut sections_sorted = true;
+        let mut last_offset = None;
 
-        Ok(C13LineProgram {
-            data,
+        let mut section_iter = DebugSubsectionIterator::new(data);
+        while let Some(sec) = section_iter.next()? {
+            match sec.kind {
+                DebugSubsectionKind::FileChecksums => {
+                    file_checksums = DebugFileChecksumsSubsection::parse(sec.data)?;
+                }
+                DebugSubsectionKind::Lines => {
+                    let lines_section = DebugLinesSubsection::parse(sec.data)?;
+                    if let Some(last_offset) = last_offset {
+                        sections_sorted &= lines_section.header.offset < last_offset;
+                    }
+                    last_offset = Some(lines_section.header.offset);
```

Comment:
This should probably use `lines_key`, as only considering `offset` is not enough for our compound sort key.
Also, not sure this is worth optimizing for anyway, as `sort_unstable_by_key` only does a linear `O(n)` scan anyways if things *are* sorted.

Reply:
Great catch, thank you. I went with your recommendation and sort always now.